pixy/consumer_test.go

```diff
@@ -0,0 +1,486 @@
+package pixy
+
+import (
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/mailgun/kafka-pixy/Godeps/_workspace/src/github.com/mailgun/log"
+	"github.com/mailgun/kafka-pixy/Godeps/_workspace/src/github.com/mailgun/sarama"
+	. "github.com/mailgun/kafka-pixy/Godeps/_workspace/src/gopkg.in/check.v1"
+)
+
+type SmartConsumerSuite struct {
+	producer *GracefulProducer
+}
+
+var _ = Suite(&SmartConsumerSuite{})
+
+func (s *SmartConsumerSuite) SetUpSuite(c *C) {
+	InitTestLog()
+	var err error
+	config := NewConfig()
+	config.ClientID = "producer"
+	config.Kafka.SeedPeers = testKafkaPeers
+	config.ChannelBufferSize = 1
+	s.producer, err = SpawnGracefulProducer(config)
+	c.Assert(err, IsNil)
+}
+
+func (s *SmartConsumerSuite) TearDownSuite(c *C) {
+	s.producer.Stop()
+}
+
+func (s *SmartConsumerSuite) TestResolveAssignments(c *C) {
+	c.Assert(resolveAssignments(nil, nil), IsNil)
+	c.Assert(resolveAssignments(nil, []string{}), IsNil)
+	c.Assert(resolveAssignments(nil, []string{"a"}), IsNil)
+	c.Assert(resolveAssignments(nil, []string{"a", "b"}), IsNil)
```

Comment:
I'm a little surprised that this result is `nil` rather than `map[string]map[int32]none{"a": nil, "b": nil}`.

I notice you don't check for a nil result when you use this function in consumer.go, which means that if it is ever called with either argument empty, `rebalance` will panic. Is that ok?